src/helpers/enrich-dataset.ts

```diff
@@ -0,0 +1,186 @@
+            additionalFields.durableUrlShapeFile = generateDurableDownloadUrl(dataset.id, siteUrl, 'shapefile');
+        }
+    }
+
+    if (datasetAttr.supportedExtensions?.includes(WFS_SERVER)) {
+        additionalFields.accessUrlWFS = ogcUrl(datasetAttr.url, 'WFS');
+    }
+
+    if (datasetAttr.supportedExtensions?.includes(WMS_SERVER)) {
+        additionalFields.accessUrlWMS = ogcUrl(datasetAttr.url, 'WMS');
+    }
+
+    dataset.properties = {
+        ...dataset.properties,
+        ...additionalFields
+    };
+
+    return dataset;
+};
+
+function generateDurableDownloadUrl(datasetId: string, siteUrl: string, fileType: FileType) {
+    const { itemId, layerId } = parseDatasetId(datasetId);
+    return `https://${siteUrl}/api/download/v1/items/${itemId}/${fileType}?layers=${layerId}`;
+}
+
+function getDatasetKeyword(dataset: HubDataset): string[] {
+    const metaKeyword = _.get(dataset, 'metadata.metadata.dataIdInfo.searchKeys.keyword');
+
+    if (metaKeyword) {
+        return metaKeyword;
+    }
+
+    const { tags, type, typeKeywords } = dataset;
+    const hasNoTags = !tags || tags.length === 0 || !tags[0]; // if tags is undefined, the tags array is empty, or tags is an empty string
+
+    if (isPage({ type, typeKeywords } as IItem) && hasNoTags) {
+        return ['ArcGIS Hub page'];
+    }
+
+    return tags;
+}
+
+function localeToLang(locale: string) {
+    return locale ? alpha2ToAlpha3Langs[locale.split('-')[0]] : '';
+}
+
+function isLayer(hubDataset: HubDataset): boolean {
+    return /_/.test(hubDataset.id);
+}
+
+function concatUrlAndPath(siteUrl: string, path: string) {
+    try {
+        return Boolean(new URL(siteUrl)) && `${siteUrl}/${path}`;
+    } catch (e) {
+        return `https://${siteUrl}/${path}`;
+    }
+}
+
+function getDatasetLicense(dataset: HubDataset): string {
+    const licenseInfo = _.get(dataset, 'licenseInfo', '');
+    const url = _.get(dataset, 'structuredLicense.url', null);
+
+    // Override hub.js default license value of 'none'
+    const license =
+        dataset.license === 'none' ?
+            null :
+            (!dataset.license || dataset.license.match(/{{.+}}/g)?.length)
+                ? (url || licenseInfo || '') :
+                dataset.license;
+
+    return license;
+}
+
+function isProxiedCSV(hubDataset: HubDataset): boolean {
+    const item = datasetToItem({
+        id: hubDataset.id,
+        attributes: hubDataset
+    } as DatasetResource);
+    const requestOptions: IHubRequestOptions = { isPortal: false };
+
+    return !!getProxyUrl(item, requestOptions);
+}
+
+function getAgoLandingPageUrl(datasetId: string, portalUrl: string) {
+    const { itemId, layerId } = parseDatasetId(datasetId);
+    let agoLandingPage = `${portalUrl}/home/item.html?id=${itemId}`;
+    if (layerId) {
+        agoLandingPage += `&sublayer=${layerId}`;
+    }
+    return agoLandingPage;
+}
+
+// HUBJS CANDIDATE
+function getDownloadLinkFn(downloadLink: string, hubDataset: any) {
```

Comment:
Doesn't this function link to the _old_ download api? I though we weren't using that anymore.

Reply:
Yes, it was used to generate link to the old download API.  But it was only being used to generate url for proxied CSV as download API does not support proxied CSV. Now, we agree on not supporting proxied CSV  in feeds, I have removed it.